src/ergw_aaa_config.erl

```diff
@@ -61,6 +64,26 @@ get_application(AppId) ->
     Apps = application:get_env(ergw_aaa, apps, #{}),
     maps:get(AppId, Apps, #{}).
 
+validate_term_cause_mapping(Opts) when is_list(Opts); is_map(Opts) ->
+    validate_options(fun validate_term_cause_mapping/2, Opts, ?DEFAULT_TERMINATION_CAUSE_MAPPING, map);
+validate_term_cause_mapping(Opts) ->
+    throw({error, {termination_cause_mapping, Opts}}).
+
+validate_term_cause_mapping(Opt, Value) when is_atom(Opt), is_integer(Value) ->
+    Value;
+validate_term_cause_mapping(Opt, Value) ->
+    throw({error, {termination_cause_mapping, {Opt, Value}}}).
+
+radius_validate_term_cause_mapping(Opts) when is_list(Opts); is_map(Opts) ->
+    validate_options(fun radius_validate_term_cause_mapping/2, Opts, ?RADIUS_DEFAULT_TERMINATION_CAUSE_MAPPING, map);
+radius_validate_term_cause_mapping(Opts) ->
+    throw({error, {termination_cause_mapping, Opts}}).
+
+radius_validate_term_cause_mapping(Opt, Value) when is_list(Opt), is_integer(Value) ->
+    Value;
+radius_validate_term_cause_mapping(Opt, Value) ->
+    throw({error, {termination_cause_mapping, {Opt, Value}}}).
+
```

Comment:
I would prefer to have the DIAMETER part in ergw_aaa_diameter and the RADIUS part in ergw_aaa_radius. Basically keep it close to the general DIAMETER and RADIUS logic.